src/query/interpreter.cpp

```diff
@@ -2734,14 +2742,17 @@ Interpreter::PrepareResult Interpreter::Prepare(const std::string &query_string,
   // an explicit transaction block.
   if (in_explicit_transaction_) {
     AdvanceCommand();
+  } else {
+    memgraph::metrics::IncrementCounter(memgraph::metrics::ActiveTransactions);
```

Comment:
How about counting active transactions as those that are explicit or those that require creating db_accessor? Then no need to modify this else if. You can put this increment in the if condition below line 2798

Reply:
Because here can actually happen that we count active transaction but don't create one